packages/typespec-ts/test/modularUnit/anonymousModel.spec.ts

```diff
@@ -0,0 +1,973 @@
+          "Record<string, any>"
+        );
+      });
+
+      it("should map empty named model(PublishResult {}) => PublishResult", async () => {
+        const tspContent = `
+        model PublishResult {
+        }
+        op read(): PublishResult;
+        `;
+        const modelFile = await emitModularModelsFromTypeSpec(tspContent);
+        // console.log(modelFile?.getFullText());
+        assertEqualContent(
+          modelFile?.getFullText()!,
+          `
+          export interface PublishResult {}
+        `
+        );
+        const operationFiles = await emitModularOperationsFromTypeSpec(
+          tspContent
+        );
+        assert.ok(operationFiles);
+        assert.equal(operationFiles?.length, 1);
+        // Model name referred in operations.ts
+        verifyReturnTypeAsEmpty(
+          operationFiles?.[0]?.getFullText()!,
+          "PublishResult"
+        );
+      });
+
+      it("should return non-empty anonymous  model({ ... })", async () => {
+        const tspContent = `
+        op read(): { foo?: {bar: string | null}};
+        `;
+        // No models.ts file generated
+        assert.isUndefined(await emitModularModelsFromTypeSpec(tspContent));
+        const operationFiles = await emitModularOperationsFromTypeSpec(
+          tspContent
+        );
+        assert.equal(operationFiles?.length, 1);
+        // Generate the operations.ts file with empty model
+        assertEqualContent(
+          operationFiles?.[0]!.getFullText()!,
+          `
+        import { TestingContext as Client } from "../rest/index.js";
+        import {
+          StreamableMethod,
+          operationOptionsToRequestParameters,
+        } from "@azure-rest/core-client";
+        export function _readSend(
+          context: Client,
+          options: ReadOptions = { requestOptions: {} }
+        ): StreamableMethod<Read200Response> {
+          return context
+            .path("/")
+            .get({ ...operationOptionsToRequestParameters(options) });
+        }
+        export async function _readDeserialize(
+          result: Read200Response
+        ): Promise<{ foo?: { bar: string | null } }> {
+          if (result.status !== "200") {
+            throw result.body;
+          }
+          return {
+            foo: !result.body.foo ? undefined : { bar: result.body.foo?.["bar"] },
+          };
+        }
+        export async function read(
+          context: Client,
+          options: ReadOptions = { requestOptions: {} }
+        ): Promise<{ foo?: { bar: string | null } }> {
+          const result = await _readSend(
+            context,
+            options
+          );
+          return _readDeserialize(result);
+        }`,
+          true
+        );
+      });
+    });
+    describe("happens as a property in response body", async () => {
+      it("should map empty anonymous  model({}) => Record<string, any> & empty named model(A {}) => Record<string, any>", async () => {
```

Comment:
I would call out for the test case where 
- We treat the anonymous empty model `{}` as `Record<string, any>` the benefit is the customers could assign objects with properties but will take pontential breaking if the contract is changed to add optional properties in e.g `{ a?: string }`